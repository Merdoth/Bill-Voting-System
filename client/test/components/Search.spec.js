import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Search } from
  '../../js/components/Search';

/* global jest */

const setup = () => {
  const props = {
    bill: [
      {
        success: true,
        message: 'bill successfully created',
        createdBill: {
          updated_at: '2018-03-27T20:54:34.393Z',
          created_at: '2018-03-27T20:54:34.393Z',
          title: 'JAMB OFFICIALS AND SNAKES',
          description: 'How a snake stole 36million from nigeria jamb who can believe is lie.',
          _id: '5abaaf8a8e736588c24978f3',
          billProgress: 'Not enacted',
          downVoteCount: 0,
          upVoteCount: 0
        },
      },
    ],
    user: {
      token: {}
    },
    result: [],
    searchBill: () => jest.fn(),
    currentPage: {},
    paginate: {
      pageCount: 1
    }
  };

  return shallow(<Search {...props} />);
};

describe('Search Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('onChange()', () => {
    it('should set search to state when input values changes', () => {
      const event = {
        preventDefault: jest.fn(),
        target: { name: 'searchTerm', value: '' }
      };
      const wrapper = setup();
      const searchInput = wrapper.find('#search');

      event.target.value = 'Senators';
      searchInput.simulate('change', event);

      expect(wrapper.instance().state.searchTerm)
        .toBe('Senators');
    });
  });
});
