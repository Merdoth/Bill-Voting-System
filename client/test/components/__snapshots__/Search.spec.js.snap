// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component snapshot it should render the right amount of elements 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      >
        <div
          className="input-field s6 search"
        >
          <label
            className="control-label"
            htmlFor="search"
          >
            Search:
          </label>
          <input
            className="form-control"
            id="search"
            name="searchTerm"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </span>
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="flexer flexer-row"
    >
      <div
        className="flexed bill-list"
      >
        <div
          className="bill-header"
        >
          <h3>
            Result
          </h3>
        </div>
        <div
          className="bill-body"
        >
          <h3>
            Sadly nothing to show :)
          </h3>
          <PaginationBoxView
            activeClassName="active"
            breakLabel="..."
            containerClassName="pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={1}
            nextClassName="next"
            nextLabel="next"
            onPageChange={[Function]}
            pageCount={0}
            pageRangeDisplayed={5}
            previousClassName="previous"
            previousLabel="previous"
            subContainerClassName="pages pagination"
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
